R/utils.R

```diff
@@ -270,3 +270,34 @@ list_call <- function(fn, x) {
 package_name <- function(env) {
   utils::packageName(env)
 }
+
+now_utc <- function() {
+  as.POSIXct(Sys.time(), tz = "UTC")
+}
+
+## Modified version of difftime to show timings
+## in ms and to a precision
+format_difftime <- function(time1, time2) {
+  z <- unclass(time1) - unclass(time2)
+  attr(z, "tzone") <- NULL
+  units <- if (all(is.na(z))) {
+    "secs"
+  } else {
+    diff <- min(abs(z), na.rm = TRUE)
+    if (!is.finite(diff) || diff <= 1) {
+      "ms"
+    } else if (diff < 60) {
+      "secs"
+    } else if (diff < 3600) {
+      "mins"
+    } else {
+      "hours"
+    }
+  }
+  time <- switch(units,
+                 ms = z * 1000,
+                 secs = z,
+                 mins = z / 60,
+                 hours = z / 3600)
+  paste(round(time, 2), units)
```

Comment:
```suggestion
  units_difftime <- if (units == "ms") "secs" else units
  z <- as.numeric(time1 - time2, units_difftime)
  if (units == "ms") {
    z <- z / 1000
  }
  sprintf("%.2f %s", z, units)
```

Somewhat simpler implementation. If you want to drop the trailing zero then process the z first to make it

```r
  paste(formatC(z, digits = 2, format = "f", drop0trailing = TRUE), units)
```